File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.py
def solution(storey):
    answer = 0
    while storey >0:
        # 1의 자리수로 올리고 내리는거 판단한 후,
        # 0이 될 때까지 카운트하기
        storey, move = divmod(storey, 10)
        if move > 5 or (move==5 and storey % 10 >=5): 
            move = 10-move
            storey +=1
        answer += move
    
    return answer
    

File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.js
function solution(storey) {
    var answer = 0;
    var remain;
    while(storey!==0){
        remain = storey%10;
        storey = Math.floor(storey/10);

        if(remain<5){
            answer+=remain;
        }
        else if(remain>5){
            answer+=(10-remain);
            storey++;
        }
        else if(remain===5){
            answer+=5;
            if((storey%10)>=5){
                storey++;
            }
        }   
    }    
    return answer;
}